python/ray/WebUI.ipynb

```diff
@@ -10,17 +10,266 @@
-    "\n",
+    "import redis\n",
+    "import sys"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
     "redis_address = os.environ[\"REDIS_ADDRESS\"]\n",
     "ray.init(redis_address=redis_address)"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "addr, port = ray.worker.global_worker.redis_address.split(\":\")\n",
+    "rc = redis.StrictRedis(host=addr, port=port, decode_responses=True, encoding='latin-1', encoding_errors='replace')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Task and Actor Information "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Remote Function Information \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fn_table = ray.global_state.function_table()\n",
+    "fn_list = []\n",
+    "for fn_id in fn_table:\n",
+    "    val = fn_table[fn_id]\n",
+    "    val[\"function_id\"] = fn_id\n",
+    "    fn_list.append(val)\n",
+    "qgrid.nbinstall(overwrite = True)\n",
+    "frame = pd.DataFrame(fn_list) \n",
+    "frame.columns = [\"DriverID\", \"Module\", \"Function\", \"FunctionID\"]\n",
+    "qgrid.show_grid(frame)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task Information "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pandas.io.json import json_normalize\n",
+    "\n",
+    "tt = ray.global_state.task_table()\n",
+    "tt_list = list(tt.values())\n",
+    "for d in tt_list:\n",
+    "    d['TaskSpec']['ReturnObjectIDs'] = [oid.hex() for oid in d['TaskSpec']['ReturnObjectIDs']]\n",
+    "    d['TaskSpec']['Args'] = [arg.hex() if isinstance(arg, ray.local_scheduler.ObjectID) else arg for arg in d['TaskSpec']['Args']]\n",
+    "\n",
+    "task_df = json_normalize(tt_list)\n",
+    "task_df.columns = [\"Local Scheduler ID\", \"State\", \"Actor Counter\", \"ActorID\", \"Arguments\", \"DriverID\", \"FunctionID\", \n",
+    "                  \"Parent Counter\", \"Parent Task ID\", \"Required CPUs\", \"Required GPUs\", \"Return Object IDs\", \"TaskID\" ]\n",
+    "qgrid.show_grid(task_df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Actor Information \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def hex_identifier(identifier):\n",
+    "    return binascii.hexlify(identifier).decode()\n",
+    "\n",
+    "actor_info = dict()\n",
+    "actors = rc.keys(\"Actor*\") \n",
+    "for actor in actors:\n",
+    "    actor_key_str = actor[len('Actor:'):]\n",
+    "    actor_key_bytes = actor_key_str.encode('latin-1')\n",
```

Comment:
let's get rid of all instances of `latin`.. ideally we wouldn't call encode/decode anywhere